textx/metamodel.py

```diff
@@ -547,43 +552,67 @@ def _current_namespace(self):
         return self.namespaces[self._namespace_stack[-1]]
 
     def model_from_str(self, model_str, file_name=None, debug=None,
-                       pre_ref_resolution_callback=None, encoding='utf-8'):
+                       pre_ref_resolution_callback=None, encoding='utf-8',
+                       **kwargs):
         """
         Instantiates model from the given string.
         :param pre_ref_resolution_callback: called before references are
                resolved. This can be useful to manage models distributed
                across files (scoping)
+        :param **kwargs additional arguments available through
+                _tx_model_params after initiating the model. The attribute
+                is set while executing pre_ref_resolution_callback (see
+                scoping.md)
         """
+        self._tx_model_param_definitions.check_params_and_raise_on_error(
```

Comment:
Sorry, looks like I forgot to comment on this in the last review. `check_params_and_raise_on_error` should just be `check_params`. Much simpler to read/write, no need to encode behavior in the name.

Reply:
no problem. if you find more stuff - comments welcome!

Reply:
done (2x)